### PR TITLE
Key generation, part 2

### PR DESCRIPTION
**Pull request by kyranjamie**
This PR adds a minimal state layer for our key management in RTK, replacing Zustand,. Discussed in https://github.com/leather-io/mono/pull/276, it relies heavily on _key origin paths_, fingerprint-prefixed derivation paths.

Note how this approach to state differs from the extension's state. There, with software wallets, mnemonic phrases are in memory, and key are derived directly from the root keychain. Ledger accounts are persisted by their account level xpub.
In this implementation, keys are always derived from the account level xpub, regardless of whether it's software or ledger. With this design, we intend only to access the mnemonic directly when either: 1. A signing operation is performed, 2. New accounts are generated.

I've put more thought into the state design than the API to consume the state, so welcome feedback as to how we should consume it in components. Reluctant to continue the pattern of `useCurrentBitcoinKey` as in extension, because of complexities dealing with `undefined` data.

- [x] Investigating slow runtime performance
  - Some operations, such as generating mnemonics and deriving child keys can be expensive, especially when rendering them all at once. This can cause the wallet to become irresponsive. There are any credible options to use workers, to take logic off the main thread. I think for now, optimisations such as memoisation, using `requestAnimationFrame` work to prevent janky behaviours, and generally deferring actions until needed is sufficient to avoid any perf bottlenecks.
- [x] Update store to reflect chain-agnostic "keychains" to avoid reuse of the term "accounts".
  - Renamed, but keeping chain-specific keystores separate as, Stacks for example, doesn't work with xpubs, rather address info
- [x] Add separate store to reflect 1:1 the user definition of accounts, and handle ordering, naming and other metadata
- [x] Create test page on Developer console
- [x] Write unit tests to ensure fingerprint and child key derivation logic is as expected
- [x] Add runtime functional app reset
- [x] Derive only account info of current network, for render perf
- [x] Replace Remember for Persist

**Demo**

https://github.com/user-attachments/assets/8e48dac9-898c-437a-a392-1da7dfcda517

### CODE REVIEW

File: apps/mobile/src/state/key-store.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        keychains: bitcoinKeychainPaths
+          .map(path => deriveKeychainDescriptor(rootKeychain, path))
+          .map(descriptor => ({ descriptor })),
+      };
+    },
+
+    removeAccount(fingerprint: string, accountIndex: number) {
+      dispatch(removesAccount({ fingerprint, accountIndex }));
+    },
+  };
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/state/key-store.ts:93]: @coderabbitai resolve

**Comment by kyranjamie** [comment]
@coderabbitai resolve

File: apps/mobile/src/app/wallet/developer-console/_layout.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Stack } from 'expo-router';
+
+export default function Layout() {
+  return (
+    <Stack>
+      <Stack.Screen name="wallet-manager" />
+    </Stack>
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/developer-console/_layout.tsx:7]: I just wanted it to go to a child page, but maybe this is the wrong way to do it?

File: apps/mobile/src/state/index.ts [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+
+const storageEnhancer = createAsyncStorageEnhancer([
+  'wallets',
+  'accounts',
+  'keychains',
+  'settings',
+]);
+
+export const store = configureStore({
+  reducer: rememberReducer(reducer),
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/state/index.ts:34]: It's not a super complicated package, so think it's okay to use even though its not very popular, it has a smaller feature set though so agree persist might be a better option

File: apps/mobile/src/state/keychains/bitcoin/bitcoin-keychains.slice.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      .addCase(userRemovesWallet, filterKeychainsToRemove(adapter.removeMany)),
+});
+
+export const { userAddsNewBitcoinAccount } = bitcoinKeychainSlice.actions;
+
+export const bitcoinKeychainSelectors = adapter.getSelectors(
+  (state: RootState) => state.keychains.bitcoin
+);
+
+function initalizeBitcoinKeychain(descriptor: string) {
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/state/keychains/bitcoin/bitcoin-keychains.slice.ts:64]: thanks

File: apps/mobile/src/app/wallet/developer-console/wallet-manager.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            paddingBottom: theme.spacing['5'] + bottom,
+            gap: theme.spacing[5],
+          }}
+        >
+          <Button title="Clear store" onPress={() => clearAllPersistedStorage()} />
+          <Button
+            title={generatingWallet ? 'Generating…' : 'Create wallet'}
+            onPress={() => {
+              setGeneratingWallet(true);
+              requestAnimationFrame(async () => {
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/developer-console/wallet-manager.tsx:37]: Tried it, but in this case doesn't run as well, attached vids.

**InteractionManager**
https://github.com/user-attachments/assets/cfa04d92-cab0-47b0-9528-4cc26e2fdf10

**requestAnimationFrame**
https://github.com/user-attachments/assets/0a793e5c-3e78-4ace-a956-98248121db5d

File: apps/mobile/src/app/wallet/developer-console/_layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/developer-console/_layout.tsx:7]: I'm reluctant to change it, as it's just a demo page which we can remove entirely when the account mgmt ui is in place.

It does get messy fast though, so also curious how we can reuse/tidy up the layouts and have consistent headers between pages.

File: apps/mobile/src/app/wallet/developer-console/index.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import { Box, Theme } from '@leather.io/ui/native';
+
+export default function DeveloperConsoleScreen() {
+  const { bottom } = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  // const contentOffsetTop = HEADER_HEIGHT;
+  const [getAddressesMessage, setGetAddressesMessage] = useState<BrowserMessage | null>(null);
+  return (
+    <>
```
Comment (pete-watters) [inline apps/mobile/src/app/wallet/developer-console/index.tsx:20]: The `<>` Fragments are not needed in a few places

File: apps/mobile/src/state/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pete-watters) [reply apps/mobile/src/state/index.ts:34]: I think we should change to `redux-persist` in future but its not mandatory yet

File: apps/mobile/src/state/keychains/keychain-state-helpers.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import {
+  extractAccountIndexFromDescriptor,
+  extractAccountIndexFromPath,
+  extractFingerprintFromDescriptor,
+} from '@leather.io/crypto';
+
+// These helpers are in a separate file owing to issues running vitest on the other file with state and react imports
```
Comment (pete-watters) [inline apps/mobile/src/state/keychains/keychain-state-helpers.ts:7]: I don't really understand the comment. Can we just remove it?

**Review by pete-watters — APPROVED** [review]
Nice work! 🚀

File: apps/mobile/src/app/wallet/developer-console/_layout.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/app/wallet/developer-console/_layout.tsx:7]: Nice, this works, thanks

File: apps/mobile/src/state/storage-persistors.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+      return SecureStore.getItemAsync(fingerprint);
+    },
+    async setMnemonic(mnemonic: string) {
+      return SecureStore.setItemAsync(fingerprint, mnemonic);
+    },
+    async deleteMnemonic() {
+      return SecureStore.deleteItemAsync(fingerprint);
+    },
+  };
+}
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/mobile/src/state/storage-persistors.ts]: Will do in next PR that requires changes to this file

**Pull request merged**